Issues/637 ambigious entity name

AndreyDegtyarev opened the pull request:
_/*Most changes are related to code generator changes
The only things that have been changed are : templates files (like DeleteByIdCommand.template) and added PropertyNameValidator
*/_

Added namespace prefix for generated entity to avoid ambiguity.
Added validator to compare property name and class name.

Commit:
Nox.Generator #637
[Nox.Generators] If an Entity is called User or Application, the compilation fails#637

Commit:
Nox.Generator #637
[Nox.Generators] If an Entity is called User or Application, the compilation fails#637

Commit:
Merge remote-tracking branch 'remotes/origin/main' into issues/637-Ambigious_Entity_Name

# Conflicts:
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.AddEmployeePhoneNumberCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateBookingCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCommissionCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCountryCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCountryHolidayCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCountryTimeZonesCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCurrencyBankNotesCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCurrencyCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCurrencyUnitsCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCustomerCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCustomerPaymentDetailsCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateCustomerTransactionCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateEmployeeCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateExchangeRateCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateHolidaysCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateLandLordCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateMinimumCashStockCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreatePaymentProviderCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateVendingMachineCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.CreateVendingMachineOrderCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteBookingByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCommissionByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCountryByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCountryHolidayByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCountryTimeZonesByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCurrencyBankNotesByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCurrencyByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCurrencyUnitsByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCustomerByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCustomerPaymentDetailsByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteCustomerTransactionByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteEmployeeByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteExchangeRateByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteHolidaysByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteLandLordByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteMinimumCashStockByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeletePaymentProviderByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteVendingMachineByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.DeleteVendingMachineOrderByIdCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateBookingCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCommissionCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCountryCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCountryHolidayCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCountryTimeZonesCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCurrencyBankNotesCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCurrencyCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCurrencyUnitsCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCustomerCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCustomerPaymentDetailsCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateCustomerTransactionCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateEmployeeCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateExchangeRateCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateHolidaysCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateLandLordCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateMinimumCashStockCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdatePaymentProviderCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateVendingMachineCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.PartialUpdateVendingMachineOrderCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateBookingCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCommissionCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCountryCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCountryHolidayCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCountryTimeZonesCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCurrencyBankNotesCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCurrencyCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCurrencyUnitsCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCustomerCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCustomerPaymentDetailsCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateCustomerTransactionCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateEmployeeCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateExchangeRateCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateHolidaysCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateLandLordCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateMinimumCashStockCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdatePaymentProviderCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateVendingMachineCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands.UpdateVendingMachineOrderCommand.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.BookingMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CommissionMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CountryHolidayMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CountryMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CountryTimeZonesMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CurrencyBankNotesMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CurrencyMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CurrencyUnitsMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CustomerMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CustomerPaymentDetailsMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.CustomerTransactionMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.EmployeeMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.EmployeePhoneNumberMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.ExchangeRateMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.HolidaysMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.LandLordMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.MinimumCashStockMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.PaymentProviderMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.VendingMachineMapper.g.cs
#	samples/Cryptocash.Api/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Factories.VendingMachineOrderMapper.g.cs
#	src/SampleWebApp/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/0.Generator.g.cs
#	src/SampleWebApp/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Infrastructure.Persistence.DtoDbContext.g.cs
#	src/SampleWebApp/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Infrastructure.Persistence.SampleWebAppDbContext.g.cs
#	src/SampleWebApp/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Presentation.Api.OData.ODataServiceCollectionExtensions.g.cs
#	tests/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/0.Generator.g.cs

ILoveMyTortoise commented on src/Nox.Solution/Validation/Domain/EntityValidator.cs:
```diff
@@ -60,23 +66,10 @@ public EntityValidator(IEnumerable<Entity>? entities, Application? application)
             RuleForEach(e => e.Events)
                 .SetValidator(e => new DomainEventValidator(domainEvents, appEvents, e.Name));
 
-            RuleForEach(p => p.Keys)
```
Why remove?

ILoveMyTortoise commented on src/Nox.Solution/ValidationResources.resx:
```diff
@@ -453,4 +453,7 @@
     <value>Attribute name '{0}' in unique attribute constraint not found in neither entity attribute(s) ({1}) nor entity key(s) ({2})</value>
   </data>
+  <data name="NameSameAsEnclosingType" xml:space="preserve">
+    <value>{0}. The member {1} of a class cannot have the same name as the class or struct, unless the member is a constructor.</value>
```
The member {1} of a class {0} cannot have the same name as the referenced entity.

Commit:
Nox.Generator #637
[Nox.Generators] If an Entity is called User or Application, the compilation fails#637
